cmake/modules/AddCIRCT.cmake

```diff
@@ -0,0 +1,43 @@
+include_guard()
+
+function(add_circt_dialect dialect dialect_namespace)
+  add_mlir_dialect(${ARGV})
+  add_dependencies(circt-headers MLIR${dialect}IncGen)
+endfunction()
+
+function(add_circt_interface interface)
+  add_mlir_interfce(${ARGV})
+  add_dependencies(circt-headers MLIR${dialect}IncGen)
+endfunction()
+
+function(add_circt_doc doc_filename command output_file output_directory)
+  add_mlir_doc(${ARGV})
+  add_dependencies(circt-doc ${output_file}DocGen)
+endfunction()
+
+function(add_circt_library name)
+  add_mlir_library(${ARGV})
```

Comment:
To me, the point of this is the have add_circt_library defer to add_llvm_library, not add_mlir_library().  is that your eventual intention?

Reply:
I guess the main point of this change was calling the right functions throughout our project, and getting our targets properly exported from `CIRCTConnfig.cmake`.  I was hoping to keep this as a largely non-functional change to how the libraries are built.

I thought replacing `add_mlir_library()` would have been a larger change, best saved for later.  It does quite a bit of stuff, a lot of which I do not quite understand yet https://github.com/llvm/llvm-project/blob/main/mlir/cmake/modules/AddMLIR.cmake#L51

Reply:
Makes sense.  It would be good to capture this in the commit comments.